### ISSUE TITLE
Escape string

### ISSUE THREAD

**Issue by shmox75** [jerry1100/electron-browser-storage#3]
I don't know but if code injection starts with ') and ends with ') here is a function to check if it's a code injection

```
function isCodeInjection( value ){
	let regex = /^.*'.*\).*\(.*'.*$/;
	
    return value !== undefined && value.match(regex) != null;
}
```

`this.setItem = (key, value) => !isCodeInjection(value) && execute(`window.${storageName}.setItem('${key}', '${value}');`);`

**Comment by shmox75** [comment]
Another method to completely get rid of code injection is to use JSON:

`this.setItem = (key, value) => execute(`window.${storageName}.setItem('${key}', 'JSON.stringify(${JSON.stringify(value)}'));`);`

this.getItem = key => execute(`JSON.parse(window.${storageName}.getItem('${key}'));`);

**Comment by jerry1100** [comment]
Hey @shmox75, thanks for the suggestion and sorry for the late reply.

I used your example and came up with the below code that I ran in my browser, and was still able to inject code. I did use `eval` as a replacement for `execute` but I think they should be similar enough.

```javascript
function setItem(key, value) {
  eval(`window.localStorage.setItem('${key}', 'JSON.stringify(${JSON.stringify(value)})');`);
}

const unsafeInput = `'); alert(1); ('`;

setItem('foo', unsafeInput);
```

Note: there was a mismatched single quote after the second `JSON.stringify()` that I moved one space to the right. Not sure if that was your intention.
